tests/components/zwave_js/test_humidifier.py

```diff
@@ -0,0 +1,951 @@
+                "endpoint": 0,
+                "property": "mode",
+                "propertyName": "mode",
+                "newValue": int(HumidityControlMode.AUTO),
+                "prevValue": int(HumidityControlMode.HUMIDIFY),
+            },
+        },
+    )
+    node.receive_event(event)
+
+    state = hass.states.get(HUMIDIFIER_ADC_T3000_ENTITY)
+    assert state.state == STATE_ON
+
+    client.async_send_command.reset_mock()
+
+    # Test off mode update from value updated event
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 68,
+            "args": {
+                "commandClassName": "Humidity Control Mode",
+                "commandClass": CommandClass.HUMIDITY_CONTROL_MODE,
+                "endpoint": 0,
+                "property": "mode",
+                "propertyName": "mode",
+                "newValue": int(HumidityControlMode.OFF),
+                "prevValue": int(HumidityControlMode.HUMIDIFY),
+            },
+        },
+    )
+    node.receive_event(event)
+
+    state = hass.states.get(HUMIDIFIER_ADC_T3000_ENTITY)
+    assert state.state == STATE_OFF
+
+    client.async_send_command.reset_mock()
+
+    # Test turning off when device is previously humidifying
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 68,
+            "args": {
+                "commandClassName": "Humidity Control Mode",
+                "commandClass": CommandClass.HUMIDITY_CONTROL_MODE,
+                "endpoint": 0,
+                "property": "mode",
+                "propertyName": "mode",
+                "newValue": int(HumidityControlMode.HUMIDIFY),
+                "prevValue": int(HumidityControlMode.OFF),
+            },
+        },
+    )
+    node.receive_event(event)
+
+    await hass.services.async_call(
+        HUMIDIFIER_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: HUMIDIFIER_ADC_T3000_ENTITY},
+        blocking=True,
+    )
+
+    assert len(client.async_send_command.call_args_list) == 1
+    args = client.async_send_command.call_args_list[0][0][0]
+    assert args["command"] == "node.set_value"
+    assert args["nodeId"] == 68
+    assert args["valueId"] == {
+        "ccVersion": 2,
+        "commandClassName": "Humidity Control Mode",
+        "commandClass": CommandClass.HUMIDITY_CONTROL_MODE,
+        "endpoint": 0,
+        "property": "mode",
+        "propertyName": "mode",
+        "metadata": {
+            "type": "number",
+            "readable": True,
+            "writeable": True,
+            "min": 0,
+            "max": 255,
+            "label": "Humidity control mode",
+            "states": {"0": "Off", "1": "Humidify", "2": "De-humidify", "3": "Auto"},
+        },
+        "value": int(HumidityControlMode.HUMIDIFY),
```

Comment:
I'm unsure why this test is failing, it is passing when I run it locally... I'm calling `node.receive_event(event)` above which should set the value of that z-wave value to `HUMIDIFY`, this is what the `vlaue` attribute on this line should be set to. Is there a better way to mock the state of a z-wave value for these tests?

Reply:
not sure why it works locally but you need to await hass.async_block_till_done after receiving the event so that the state change can be processed

Reply:
Added

Reply:
sorry I misread, I thought you were trying to process a state change in HA, but the issue is the node's state.

Out of curiosity, locally are you using the pip installed latest version of `zwave-js-server-python` or are you using a local version of `master`?

Reply:
I see this is also part of the diff
```
E         +  'newValue': 1,
E         +  'prevValue': 0,
```
which matches keys that are in the `event` above:
```
                "newValue": int(HumidityControlMode.HUMIDIFY),
                "prevValue": int(HumidityControlMode.OFF),
```
but locally when I run the test these keys are not even set in the dict being compared on L186...

Reply:
> Out of curiosity, locally are you using the pip installed latest version of zwave-js-server-python or are you using a local version of master?

Still using a local version since I started this branch before my changes were merged... I might need to rebase to pick up a version that has the humidity control changes?